### PR TITLE
[Opt](performance) opt the outer join for nested loop join

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

a3ae5ed6-0804-4737-8963-e2c5a90d256f

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

File: be/src/vec/exec/join/vnested_loop_join_node.cpp [hunk 1]
```diff
@@ -232,6 +231,9 @@ Status VNestedLoopJoinNode::sink(doris::RuntimeState* state, vectorized::Block*
 
 Status VNestedLoopJoinNode::push(doris::RuntimeState* state, vectorized::Block* block, bool eos) {
     COUNTER_UPDATE(_probe_rows_counter, block->rows());
+    _cur_probe_row_visited_flags.resize(block->rows());
```
Comment (Gabriel39) [inline be/src/vec/exec/join/vnested_loop_join_node.cpp:234]: Use `_cur_probe_row_visited_flags.resize(block->rows(), 0);` to replace `std::fill`

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**